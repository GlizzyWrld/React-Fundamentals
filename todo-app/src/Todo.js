function Todo({ task, removeTodo }) {
  return (
    <div className="todo">
      <div>{task}</div>
      <button onClick={removeTodo}>X</button>
    </div>
  );
}

export default Todo;
