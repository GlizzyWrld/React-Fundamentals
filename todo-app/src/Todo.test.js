import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

test('renders Todo component without errors', () => {
    const { container } = render(<Todo task="Test Task" removeTodo={() => {}} />);
    expect(container).toMatchSnapshot();
});

test('calls removeTodo function when X button is clicked', () => {
    const removeTodoMock = jest.fn();
    const { container, getByText } = render(
      <Todo task="Test Task" removeTodo={removeTodoMock} />
    );
    const removeButton = getByText('X');
  
    fireEvent.click(removeButton);
  
    expect(container).toMatchSnapshot();
});
