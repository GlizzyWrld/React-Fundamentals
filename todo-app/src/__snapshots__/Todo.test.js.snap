// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls removeTodo function when X button is clicked 1`] = `
<div>
  <div
    class="todo"
  >
    <div>
      Test Task
    </div>
    <button>
      X
    </button>
  </div>
</div>
`;

exports[`renders Todo component without errors 1`] = `
<div>
  <div
    class="todo"
  >
    <div>
      Test Task
    </div>
    <button>
      X
    </button>
  </div>
</div>
`;
